webpack/assets/javascripts/react_app/routes/HOC.js

```diff
@@ -0,0 +1,61 @@
+import React, { useEffect } from 'react';
```

Comment:
This file contains really nice HOCs :+1: 

I'm wondering how the documentation part of it should be handled, especially when there are no use cases yet.
I believe adding usage examples in the comments would do the job.
Another option is to update the storybook with some examples.
Or move it to another PR that use this infrastructure.